flocker/node/agents/cinder.py

```diff
@@ -561,7 +599,33 @@ def compute_instance_id(self):
             COMPUTE_INSTANCE_ID_NOT_FOUND(
                 local_ips=local_ips, api_ips=api_ip_map
             ).write()
-            raise UnknownInstanceID(self)
+
+    def compute_instance_id(self):
+        """
+        Attempt to retrieve node UUID from the metadata in a config drive or
+        from the metadata service.
+        Fall back to finging the ``ACTIVE`` Nova API server with an
+        intersection of the IPv4 and IPv6 addresses on this node.
+        """
+        metadata_checkers = [
```

Comment:
Do you expect more checkers?

I think two calls will look simpler than loop over vector of lambdas.